util/gpool/spmc/spmcpool_test.go

```diff
@@ -114,10 +114,49 @@ func TestStopPool(t *testing.T) {
 	}()
 	// Waiting task finishing
 	control.Stop()
-	close(exit)
 	control.Wait()
-	// it should pass. Stop can be used after the pool is closed. we should prevent it from panic.
+	wg.Wait()
+	// close pool
+	pool.ReleaseAndWait()
+}
+
+func TestStopPoolWithSlice(t *testing.T) {
+	type ConstArgs struct {
+		a int
+	}
+	myArgs := ConstArgs{a: 10}
+	// init the pool
+	// input type， output type, constArgs type
+	pool, err := NewSPMCPool[int, int, ConstArgs, any, pooltask.NilContext]("TestStopPoolWithSlice", 3, rmutil.UNKNOWN)
+	require.NoError(t, err)
+	pool.SetConsumerFunc(func(task int, constArgs ConstArgs, ctx any) int {
+		return task + constArgs.a
+	})
+
+	exit := make(chan struct{})
+
+	pfunc := func() ([]int, error) {
+		select {
+		case <-exit:
+			return nil, gpool.ErrProducerClosed
+		default:
+			return []int{1, 2, 3}, nil
+		}
+	}
+	// add new task
+	resultCh, control := pool.AddProduceBySlice(pfunc, myArgs, pooltask.NilContext{}, WithConcurrency(4))
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
```

Comment:
Could we add the `control.Stop()` here, like https://github.com/pingcap/tidb/pull/40542/files#diff-192b5bb5d7cd86dfa5e5223cdece209444dc6b5de220d92648d4385098228b9bR183-R190

Reply:
Done